arch/riscv/core/irq_offload.c

```diff
@@ -33,13 +33,8 @@ void z_irq_do_offload(void)
 
 void arch_irq_offload(irq_offload_routine_t routine, const void *parameter)
 {
-	unsigned int key;
-
-	key = irq_lock();
```

Comment:
Also because irq_lock() is a complicated global recursive spinlock in SMP that no one should be using anymore.  For the rare circumstances where you want to mask interrupts on the local CPU, we can still use arch_irq_lock().  Synchronization should be done with k_spin_lock() for all new code.